### PR TITLE
Get user info

### LINKED ISSUE #71: Pull in user information into rogue using Northstar or Cached data
_FEATURE OVERVIEW_
-------------------
#### User Story

Eventually we will want to be able to expose important user information in rogue. To do this we should implement a system in rogue that when wanting to display user information will first look in Cache and grab the user info from there. If it isn't found there, make the call to Northstar to grab the user info and also store that information in cache so that the next time we need that users information it will be there (or not be there depending on when we expire the cache.)

#### Why This Matters

When we start to build out the admin interface for Rogue we will want to be able to display things like user name, email, address, etc.

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

Creates the system of grabbing user information in Rogue. 

First we try to grab user information that we stored in cache, if they aren't found there we grab them directly from Northstar. 

This work is based largely on @weerd 's work in Gladiator. The Gladiator system is a little more complicated since we actually store users in Gladiator, but in Rogue we just store references to northstar ids in the reportbacks table. 

So what this sets up is a `Registrar` service that handles the logic of resolving users in northstar vs cache. We also have a `CacheRepository` class that handles the storing/grabbing cache logic [(based on this)](https://github.com/DoSomething/gladiator/blob/master/app/Repositories/CacheStorage.php) 

This also sets up a view at `/users` to just list the users in the users table, which will really only be admins but helps to provide an interface for this work. 

#### Any background context you want to provide?

I started writing tests for this, but decided I needed to get this merged in more. I will flag that work for the future. 

#### Relevant tickets
Fixes #71 

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love.

### CODE REVIEW

File: app/Services/Registrar.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    public function find($id)
+    {
+        $user = $this->cache->retrieve($id);
+
+        if (! $user) {
+            $user = $this->northstar->getUser('id', $id);
+
+            // @TODO - How long should we store users in Cache?
+            $this->cache->store($user->id, $user);
```
Comment (angaither) [inline app/Services/Registrar.php]: can we push updates here so it's only updated when it needs to be?

File: app/helpers.php [hunk 2]
```diff
@@ -28,9 +28,14 @@ function edit_image($image, $coords)
...
-    return $newTransactionId;
+    if ($transactionId) {
+        $transactionIdParts = explode('-', $transactionId);
+        $incrementedStep = $transactionIdParts[1] + 1;
+        $newTransactionId = $transactionIdParts[0] . '-' . $incrementedStep;
+
+        return $newTransactionId;
+    }
+
+    return null;
```
Comment (angaither) [inline app/helpers.php]: should this return null if there isn't a transaction id? or should it create one?

File: app/Http/Controllers/UsersController.php [hunk 3]
```diff
@@ -12,8 +13,22 @@ class UsersController extends Controller
...
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $ids = User::all()->pluck('northstar_id')->all();
```
Comment (angaither) [inline app/Http/Controllers/UsersController.php]: this probably needs a pager, rather than displaying all users

Comment (sbsmith86) [inline app/Http/Controllers/UsersController.php]: I could see a need for that in the future. But this is only a page that displays users in the `Users` table which is really only admins, which is a small set of people right now. I mainly put this here to just have a visual display in the app of getting user information.

File: app/Services/Registrar.php (hunk 1 above)
Comment (sbsmith86) [inline app/Services/Registrar.php]: I'm not sure what you mean? Like only store the user if they have updated information?

File: app/Http/Controllers/UsersController.php (hunk 3 above)
Comment (angaither) [inline app/Http/Controllers/UsersController.php]: oh my bad, I thought this was like gladiator, where we had to kill the whole user page

File: app/helpers.php (hunk 2 above)
Comment (sbsmith86) [inline app/helpers.php]: hmm ya i guess we could create one instead. The reason I did this was mainly for testing. When trying to test the API, the tests were failing cause we don't send over a transaction ID with the fake requests. But I could see an instance where requests are coming in without the header for reals. But in those cases, would we want rogue to be responsible for creating the header here? Or would we want to not accept the request without that header?

**Comment by DFurnes** [comment]
Looks good to me! 👍 This makes me want to bundle some caching logic into Gateway in the future as well. Would be nice to have all of that standardized between apps.

File: app/Services/Registrar.php (hunk 1 above)
Comment (sbsmith86) [inline app/Services/Registrar.php]: oooo hmm yea. I like that idea! I'm not clear on how we can tell in Gateway that a user has updated information.

File: app/helpers.php (hunk 2 above)
Comment (angaither) [inline app/helpers.php]: Yeah, I'm wondering about this for the migration script as well...

**Pull request merged**